### PR TITLE
python3Packages.pythonix: 0.1.6 -> 0.1.7, fix python module

### PR DESCRIPTION
**Pull request by r-burns**
Pythonix uses the same python as meson, so we need to override
meson to use the right python version or the module won't import.

Also added a pythonImportsCheck to prevent regressions.

Corresponding upstream fix: https://github.com/Mic92/pythonix/commit/ee22a8f732344d802798e31ce73d1fcd016e8aed

###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/103151#issuecomment-723638622

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-burns** [comment]
Result of `nixpkgs-review` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.nixpkgs</li>
    <li>python37Packages.pythonix</li>
    <li>python38Packages.nixpkgs</li>
    <li>python38Packages.pythonix</li>
    <li>python39Packages.nixpkgs</li>
    <li>python39Packages.pythonix</li>
  </ul>
</details>

File: pkgs/development/python-modules/pythonix/default.nix [hunk 1]
```diff
@@ -18,6 +18,10 @@ buildPythonPackage rec {
 
   buildInputs = [ nix boost ];
 
+  postInstall = ''
+    PYTHONPATH=$out/${python.sitePackages}:$PYTHONPATH python -c "import nix"
```
Comment (Mic92) [inline pkgs/development/python-modules/pythonix/default.nix:22]: Afaik the same as: 

```
pythonImportsCheck = [ "nix" ];
```

Comment (r-burns) [reply pkgs/development/python-modules/pythonix/default.nix:22]: Yeah I had tried that but oddly I get this error:
```
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: nix
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/m7080pw0ryjk0jhljp55rq1hd2qy8gki-python3-3.8.6/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'nix'
```
And yet the manual check works, as well as the resulting package in a nix-shell. Maybe pythonImportsCheck doesn't work with the meson build?

Comment (Mic92) [reply pkgs/development/python-modules/pythonix/default.nix]: Mhm. Maybe I miss some python metadata that importlib expects?

File: pkgs/development/python-modules/pythonix/default.nix [hunk 2]
```diff
@@ -18,6 +18,13 @@ buildPythonPackage rec {
 
   buildInputs = [ nix boost ];
 
+  postInstall = ''
+    # XXX why does pythonImportsCheck not set this?
+    export PYTHONPATH=$out/${python.sitePackages}:$PYTHONPATH
```
Comment (Mic92) [inline pkgs/development/python-modules/pythonix/default.nix:23]: @FRidh @jonringer Why is this line necessary? This build install a python modules with meson. Is there something else necessary. The python module is a shared-library only, no python source code.

Comment (r-burns) [reply pkgs/development/python-modules/pythonix/default.nix:24]: Ah, I think I figured it out. The pythonpath is typically set by the pipInstallHook or eggInstallHook. So when we use e.g. a cmake or meson install the package isn't added to the pythonpath. I like your approach much better, so I just updated the comment

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -6007,7 +6007,10 @@ in {
 
   pythonIRClib = callPackage ../development/python-modules/pythonirclib { };
 
-  pythonix = callPackage ../development/python-modules/pythonix { inherit (pkgs) meson pkgconfig; };
+  pythonix = callPackage ../development/python-modules/pythonix {
+    inherit (pkgs) pkgconfig;
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6011]:
```suggestion
    inherit (pkgs) pkg-config;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (r-burns) [reply pkgs/top-level/python-packages.nix]: Ok, done

File: pkgs/development/python-modules/pythonix/default.nix (hunk 2 above)
Comment (FRidh) [reply pkgs/development/python-modules/pythonix/default.nix:24]: May want to create a `pyproject.toml` for the project that uses `mesonpep517` as build backend. Then the backend will use meson to build a wheel and we install the wheel using pip.
https://github.com/FRidh/mesonpep517examples

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108259` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.nixpkgs</li>
<li>python37Packages.pythonix</li>
<li>python38Packages.nixpkgs</li>
<li>python38Packages.pythonix</li>
<li>python39Packages.nixpkgs</li>
<li>python39Packages.pythonix</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.nixpkgs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.nixpkgs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.nixpkgs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108259` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.nixpkgs</li>
<li>python37Packages.pythonix</li>
<li>python38Packages.nixpkgs</li>
<li>python38Packages.pythonix</li>
<li>python39Packages.nixpkgs</li>
<li>python39Packages.pythonix</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.nixpkgs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.nixpkgs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.nixpkgs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Review by rmcgibbo — APPROVED** [review]

File: pkgs/development/python-modules/pythonix/default.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/development/python-modules/pythonix/default.nix:24]: I need to wait for upstream to fix this bug so: https://gitlab.com/thiblahute/mesonpep517/-/issues/5
